### PR TITLE
ci: Transition GARM tests to free runners, pt. III

### PR DESCRIPTION
**Pull request by sprt**
Follow up to #9960 and #10007.

Tracked in #9940.

### CODE REVIEW

File: .github/workflows/static-checks.yaml [hunk 1]
```diff
@@ -66,10 +64,6 @@ jobs:
           sudo -E ./ci/install_yq.sh
         env:
           INSTALL_IN_GOPATH: false
-      - name: Install rust
```
Comment (wainersm) [inline .github/workflows/static-checks.yaml:69]: Although rust is preinstalled in the free runner, I think we should stick with version in https://github.com/kata-containers/kata-containers/blob/main/versions.yaml#L374

File: .github/workflows/PR-wip-checks.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ concurrency:
 
 jobs:
   pr_wip_check:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```
Comment (wainersm) [inline .github/workflows/PR-wip-checks.yaml:18]: Gods of versions pinned will love it. thanks @sprt !

File: .github/workflows/static-checks.yaml (hunk 1 above)
Comment (sprt) [reply .github/workflows/static-checks.yaml:69]: Agreed, will do this.

Comment (sprt) [reply .github/workflows/static-checks.yaml]: Done

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks @sprt

**Review by GabyCT — APPROVED** [review]

**Pull request merged**